### PR TITLE
Modify the display of the slash screen

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

https://github.com/cocos/3d-tasks/issues/18305#issuecomment-2205138839

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/game/splash-screen.ts [hunk 1]
```diff
@@ -37,13 +37,15 @@ import { SetIndex } from '../rendering/define';
...
 import { PipelineRuntime } from '../rendering/custom';
+import { ResolutionPolicy } from '../ui/view';
 
 const v2_0 = new Vec2();
 type SplashLogoType = 'default' | 'none' | 'custom';
 type SplashBackgroundType = 'default' | 'color' | 'custom';
 type WatermarkLocationType = 'default' | 'topLeft' | 'topRight' | 'topCenter' | 'bottomLeft' | 'bottomCenter' | 'bottomRight';
 
 interface ISplashSetting {
+    policy?: number;
```
Comment (minggo) [inline cocos/game/splash-screen.ts:48]: Why the type is not ResolutionPolicy?

File: cocos/game/splash-screen.ts [hunk 2]
```diff
@@ -125,7 +126,15 @@ export class SplashScreen {
     }
 
     public init (): Promise<void[]> {
+        let policy: number = ResolutionPolicy.SHOW_ALL;
+        if (!EDITOR) {
+            const designResolution = settings.querySettings(Settings.Category.SCREEN, 'designResolution');
+            if (designResolution != null) {
```
Comment (minggo) [inline cocos/game/splash-screen.ts]: Use !== instead?

File: cocos/game/splash-screen.ts (hunk 1 above)
Comment (qiuguohua) [reply cocos/game/splash-screen.ts:48]: ResolutionPolicy is an object, but the value inside the setting is a number

File: cocos/game/splash-screen.ts (hunk 2 above)
Comment (qiuguohua) [reply cocos/game/splash-screen.ts]: OK

Comment (minggo) [reply cocos/game/splash-screen.ts:132]: ok

File: cocos/game/splash-screen.ts [hunk 3]
```diff
@@ -307,12 +316,31 @@ export class SplashScreen {
...
+                scaleX = (dh / this.bgImage.height) * this.bgImage.width;
                 scaleY = dh;
+            } else if (this.settings.policy === ResolutionPolicy.SHOW_ALL) {
+                if (dw > dh) {
+                    scaleX = (dh / this.bgImage.height) * this.bgImage.width;
+                    scaleY = dh;
+                } else {
+                    scaleX = dw;
+                    scaleY = (dw / this.bgImage.width) * this.bgImage.height;
+                }
```
Comment (minggo) [inline cocos/game/splash-screen.ts:332]: The logic seems not correct, should choose the one the has small scale.

Comment (qiuguohua) [reply cocos/game/splash-screen.ts:332]: Fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**